prov-api/helpers/generate_report_helpers.py

```diff
@@ -0,0 +1,223 @@
+import io
+import os
+from typing import Dict, List, Any, Callable, ByteString
+
+from pydantic import ValidationError 
+from config import Config
+from fastapi import HTTPException
+
+from ProvenaInterfaces.ProvenanceAPI import LineageResponse
+from ProvenaInterfaces.RegistryAPI import ItemBase, ItemSubType, SeededItem, Node
+from helpers.entity_validators import RequestStyle, validate_model_run_id, validate_study_id
+from helpers.neo4j_helpers import upstream_query, downstream_query
+
+from tempfile import NamedTemporaryFile
+
+from docx import Document
+from docx.shared import Inches
+
+async def validate_node_id(node_id: str, item_subtype: ItemSubType, request_style: RequestStyle, config: Config) -> None: 
+    """Validates that a provided node (id + subtype) does exist within the registry.
+
+    Parameters
+    ----------
+    node_id : str
+        The ID of the node.
+    item_subtype : ItemSubType
+        The subtype of the node.
+    request_style : RequestStyle
+        The style of the request (user, service account etc)
+    config : Config
+        A config object containing information about the different endpoints of the system.
+
+    Raises
+    ------
+    HTTPException
+        If an invalid subtype is provided.
+    HTTPException
+        If fetched entity contains an error this exception is raised.
+    HTTPException
+        If the fetched entity is of type SeededItem this exception is raised.
+    """
+    
+    validation_mapping: Dict[ItemSubType, Callable] = {
+        ItemSubType.MODEL_RUN: validate_model_run_id,
+        ItemSubType.STUDY: validate_study_id
+    }
+
+    validator_func = validation_mapping.get(item_subtype)
+
+    if not validator_func:
+        raise HTTPException(status_code=400, detail="Unsupported item subtype for validation.")
```

Comment:
Nice i like this approach. 

I think the error message could be enhanced to be something like "Unsupported item subtype for validation. Received subtype X. Supported types are [y,z]". You could get the supported types dynamically from the keys in the validation mapping dict defined above aswell. Its a bit longer message but can help with debugging.

Reply:
Changed - Thanks Ross!